src/main/java/de/zalando/aruha/nakadi/service/CursorsCommitService.java

```diff
@@ -0,0 +1,50 @@
+package de.zalando.aruha.nakadi.service;
+
+import de.zalando.aruha.nakadi.domain.Cursor;
+import de.zalando.aruha.nakadi.exceptions.NoSuchSubscriptionException;
+import de.zalando.aruha.nakadi.exceptions.ServiceUnavailableException;
+import de.zalando.aruha.nakadi.repository.TopicRepository;
+import de.zalando.aruha.nakadi.repository.zookeeper.ZooKeeperHolder;
+import de.zalando.aruha.nakadi.repository.zookeeper.ZooKeeperLockFactory;
+
+import java.nio.charset.Charset;
+
+import static de.zalando.aruha.nakadi.repository.zookeeper.ZookeeperUtils.runLocked;
+import static java.text.MessageFormat.format;
+
+public class CursorsCommitService {
+
+    private static final Charset CHARSET = Charset.forName("UTF-8");
+
+    private final ZooKeeperHolder zkHolder;
+    private final TopicRepository topicRepository;
+    private final ZooKeeperLockFactory zkLockFactory;
+
+    public CursorsCommitService(final ZooKeeperHolder zkHolder, final TopicRepository topicRepository,
+                                final ZooKeeperLockFactory zkLockFactory) {
+        this.zkHolder = zkHolder;
+        this.topicRepository = topicRepository;
+        this.zkLockFactory = zkLockFactory;
+    }
+
+    public boolean commitCursor(final String subscriptionId, final String eventType, final Cursor cursor)
+            throws ServiceUnavailableException, NoSuchSubscriptionException {
+
```

Comment:
After discussion with @antban it was decided that it makes sense to make commitCursors method that gets a batch of cursors and it will do validation and will call commitCursor for each cursor.